tests/invalid_models_tests/test_models.py

```diff
@@ -761,3 +761,69 @@ class ParkingLot(Place):
             'as an implicit link is deprecated.'
         )
         self.assertEqual(ParkingLot._meta.pk.name, 'parent')
+
+    def test_m2m_table_name_clash(self):
+        class Foo(models.Model):
+            bar = models.ManyToManyField('Bar', db_table='myapp_bar')
+
+            class Meta:
+                db_table = 'myapp_foo'
+
+        class Bar(models.Model):
+            class Meta:
+                db_table = 'myapp_bar'
+
+        errors = Foo.check()
+        expected = [
+            Error(
+                "Field is using a table that has already been registered by "
+                "'invalid_models_tests.Bar.bar'.",
```

Comment:
`Bar.bar` isn't a field. The message should be `Field is using a table that has already been registered by 'invalid_models_tests.Bar'` as the `db_table` is declared on a non-`auto_created` model (`_meta.auto_created is False`).